test/functional/basic/basic-e2e-specs.js

```diff
@@ -202,8 +203,12 @@ describe('XCUITestDriver - basics', function () {
     });
 
     it('should start a session, navigate to url, get title', async () => {
-      let contexts = await driver.contexts();
-      contexts.length.should.be.at.least(2);
+      let contexts;
+      await retryInterval(10, 1000, async () => {
+        // on some systems (like Travis) it takes a while to load the webview
+        contexts = await driver.contexts();
+        contexts.length.should.be.at.least(2);
```

Comment:
Not sure what you mean. This will call up to 10 times, waiting a second in between, until nothing in the inner function throws an error (i.e. when there are at least 2 contexts). If after the 10 tries there is still no second context, something is wrong and it will fail.